slack_sdk/models/blocks/block_elements.py

```diff
@@ -1381,6 +1405,185 @@ def __init__(
+                You can use this when you receive a view_submission payload to identify the value of the input element.
+                Should be unique among all other action_ids in the containing block.
+                Maximum length for this field is 255 characters.
+            initial_value: The initial value in the URL input when it is loaded.
+            dispatch_action_config: A dispatch configuration object that determines when during text input
+                the element returns a block_actions payload.
+            focus_on_load: Indicates whether the element will be set to auto focus within the view object.
+                Only one element can be set to true. Defaults to false.
+            placeholder: A plain_text only text object that defines the placeholder text shown in the URL input.
+                Maximum length for the text in this field is 150 characters.
+        """
+        super().__init__(
+            type=self.type,
+            action_id=action_id,
+            placeholder=TextObject.parse(placeholder, PlainTextObject.type),
+            focus_on_load=focus_on_load,
+        )
+        show_unknown_key_warning(self, others)
+
+        self.initial_value = initial_value
+        self.dispatch_action_config = dispatch_action_config
+
+
+# -------------------------------------------------
+# Number Input Element
+# -------------------------------------------------
+
+
+class NumberInputElement(InputInteractiveElement):
+    type = "number_input"
+
+    @property
+    def attributes(self) -> Set[str]:
+        return super().attributes.union(
+            {
+                "initial_value",
+                "is_decimal_allowed",
+                "min_value",
+                "max_value",
+                "dispatch_action_config",
+            }
+        )
+
+    def __init__(
+        self,
+        *,
+        action_id: Optional[str] = None,
+        is_decimal_allowed: Optional[bool] = None,
+        initial_value: Optional[str] = None,
+        min_value: Optional[str] = None,
+        max_value: Optional[str] = None,
+        dispatch_action_config: Optional[Union[dict, DispatchActionConfig]] = None,
+        focus_on_load: Optional[bool] = None,
+        placeholder: Optional[Union[str, dict, TextObject]] = None,
+        **others: dict,
+    ):
+        """
+        https://api.slack.com/reference/block-kit/block-elements#number
+
+        Args:
+            action_id (required): An identifier for the input value when the parent modal is submitted.
+                You can use this when you receive a view_submission payload to identify the value of the input element.
+                Should be unique among all other action_ids in the containing block.
+                Maximum length for this field is 255 characters.
+            is_decimal_allowed: Decimal numbers are allowed if is_decimal_allowed= true, set the value to false otherwise.
```

Comment:
```suggestion
            is_decimal_allowed (required): Decimal numbers are allowed if is_decimal_allowed= true, set the value to false otherwise.
```

Reply:
Good catch :100: